### PR TITLE
[fga] enable all writes by feature flag

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description

We need to enable/disable all writes for spicedb not only the initial migration, because the relationships would run out of sync if we don't continue to update the relationships as a user uses the product.

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at 236dc7c</samp>

Refactor and simplify FGA-related code in the server component. Use consistent and clear function names for FGA checks and writes. Reduce feature flag calls and imports.

</details>

## Related Issue(s)

Fixes #

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - se-featured2b5cae174</li>
	<li><b>🔗 URL</b> - <a href="https://se-featured2b5cae174.preview.gitpod-dev.com/workspaces" target="_blank">se-featured2b5cae174.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - se-feature-flag-writes-gha.16576</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-se-featured2b5cae174%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-core-dev" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/server/src/authorization/authorizer.ts [hunk 1]
```diff
@@ -513,15 +509,23 @@ export class Authorizer {
...
     });
 }
 
+export async function isFgaWritesEnabled(userId: string): Promise<boolean> {
+    const result = await getExperimentsClientForBackend().getValueAsync("spicedb_relationship_updates", false, {
+        user: {
+            id: userId,
+        },
+    });
+    return result || isFgaChecksEnabled(userId);
```
Comment (geropl) [inline components/server/src/authorization/authorizer.ts:526]: I had to check twice here, because of the expected await. But should work fine :+1:

**Review by geropl — APPROVED** [review]
Code LGTM

File: components/server/src/authorization/authorizer.ts (hunk 1 above)
Comment (svenefftinge) [reply components/server/src/authorization/authorizer.ts:526]: Thanks, I will update it to make it clear

**Pull request merged**